### ISSUE TITLE
Feature Request: Deploy and Retrieve Apex Classes to an org

### ISSUE THREAD

**Issue by LeviticusNelson** [tzmfreedom/rustforce#25]
The way that I think that deploy could be done is to:
1. compile the files into base64 
2. put the encoded base64 string into a package.xml file
3. zip the file
4. POST /services/data/vXX.X/metadata/deployRequest

Once the deploy feature is complete it should be pretty trivial to retrieve.

References:
[Deploying and Retrieving Metadata with the Zip File](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/file_based_zip_file.htm)
[Deploy to Org using SOAP API](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_deploying_additional.htm)
[Metadata for Apex Classes](https://developer.salesforce.com/docs/atlas.en-us.238.0.api_meta.meta/api_meta/meta_classes.htm)

**Comment by tzmfreedom** [comment]
Thank you for your feature request.
I'm glad to have the opportunity to consider new features.

There is several deployment tool for salesforce, for example SFDX, JSforce.
I'd like to understand the motivation to add deployment feature for rustforce and the current issue for their deployment tool.

**Comment by LeviticusNelson** [comment]
My reasoning for implementing this feature would be to have an alternative development tool that would have the speed and safety of rust. I use SFDX as my main development tool and one of the main bottlenecks is the speed at which it operates. Seeing that the REST API alone reduces the time for certain requests, my theory that implementing a tooling API in rust using the REST API would dramatically increase the deployment process.

Another thing is that I am implementing an SFDX-like TUI in rust and it would be nice to have a crate that has this feature implemented since this crate already implements the client connection.
